qiskit/utils/deprecation.py

```diff
@@ -14,7 +14,173 @@
 
 import functools
 import warnings
-from typing import Any, Callable, Dict, Optional, Type
+from typing import Any, Callable, Dict, Optional, Type, Tuple, Union
+
+
+def deprecate_func(
+    *,
+    since: str,
+    additional_msg: Optional[str] = None,
+    pending: bool = False,
+    project_name: str = "Qiskit Terra",
```

Comment:
I agree. `qiskit-terra`. At least until with synchronise versions with `qiskit`.

Reply:
Good call to use `qiskit-terra`. The PyPI package name is much more precise. Fixed. Thanks!

Reply:
No, I don't think this would help. We still need the same way to auto-detect or allow you to set the value once for the whole application; https://github.com/Qiskit/qiskit-terra/issues/9682 remains necessary.

Reply:
Ok, I had forgotten the other PR to look at it for improvement. Hopefully the fact that's it not a custom string now, but the package name, makes it easier